cirq-core/cirq/transformers/gauge_compiling/gauge_compiling.py

```diff
@@ -154,22 +152,41 @@ def __call__(self, prng: np.random.Generator) -> Gauge:
 
 @transformer_api.transformer
 class GaugeTransformer:
+
     def __init__(
         self,
         # target can be either a specific gate, gatefamily or gateset
         # which allows matching parametric gates.
         target: Union[ops.Gate, ops.Gateset, ops.GateFamily],
         gauge_selector: Callable[[np.random.Generator], Gauge],
+        symbolize_2_qubit_gate_fn: Optional[
+            Callable[
+                [ConstantGauge, Tuple[sympy.Symbol]], Tuple[ops.Gate, Dict[str, Union[float, int]]]
+            ]
+        ] = None,
     ) -> None:
         """Constructs a GaugeTransformer.
 
         Args:
             target: Target two-qubit gate, a gate-family or a gate-set of two-qubit gates.
             gauge_selector: A callable that takes a numpy random number generator
                 as an argument and returns a Gauge.
+            symbolize_2_qubit_gate_fn: A callable that accepts a ConstantGauge and a tuple of
+              Symbol objects, returning a tuple containing the symbolized gate and a dictionary
+              mapping symbol names to their corresponding parameter values. This function is
+              used to parameterize two-qubit gates within the as_sweep method. If no function
+              is provided, no parameterization occurs. It's assumed that all gauges associated
+              with a given transformer employ the same symbolization function for their respective
+              two-qubit gates.
         """
         self.target = ops.GateFamily(target) if isinstance(target, ops.Gate) else target
         self.gauge_selector = gauge_selector
+        self.symbolize_2_qubit_gate_fn = symbolize_2_qubit_gate_fn
+        self.n_2_qubit_gate_symbols = (
```

Comment:
I don't like this `inspect` logic ... can we instead of passing a function create a class?

```py3
@frozen
class TwoQubitGateSymbolizer:
    ...

    def __call__(self, gauge: ConstantGauge):
         ...
```

Reply:
This PR already consist a couple of commits in it, shall we modulize TwoQubitGateSymbolizer in a followup cl instead?

Reply:
no ... the class will change the api since istead of expecting a function, we will expect an instance of the class

Reply:
okay, will work on this.

Reply:
thanks